### PR TITLE
PEG Grammer, Compiler and Parser

### PR DESCRIPTION
**Pull request by merkste**
Provide the PEG ASCII syntax definition accompanied by a compiler and a parser to PetitParser2.

### CODE REVIEW

**Comment by merkste** [comment]
I am a bit puzzled, as many tests fail with:

    ..RETRY->BaselineOfPetitParser2Core
    ...RETRY->BaselineOfPetitParser2Core
    ...FAILED->BaselineOfPetitParser2CoreCould not resolve: BaselineOfPetitParser2Core 
    [BaselineOfPetitParser2Core] in /Users/travis/smalltalkCI-master/_builds/pharo-local/package-cache 
    filetree:///Users/travis/build/kursjan/petitparser2

Did I mess something up with the new package or does the branch trigger an new issue with the CI checks?

**Comment by kursjan** [comment]
Nice one!

I didn't see any tests, are there some available?

WRT to the failing builds: Seems the new fileformat (tonel) is not supported by Pharo5, Pharo6. I updated the configuration and re-run the tests.

**Comment by merkste** [comment]
Hi, tests are implemented now. It took some time to polish them. =)

File: PetitParser2-PEG/PP2PEGGrammarTest.class.st [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+]
+
+{ #category : #'tests nonterminals' }
+PP2PEGGrammarTest >> testChoice [
+
+	| grammar parser |
+	grammar := 'Grammar <- "Foo" / "Bar" / "Baz"'.
+	parser := self grammar: grammar start: 'Grammar'.
+	self
+		assert: ((parser parse: '') isKindOf: PP2Failure);
```
Comment (kursjan) [inline PetitParser2-PEG/PP2PEGGrammarTest.class.st]: assert: fail:

File: PetitParser2-PEG/PP2PEGGrammarTest.class.st [hunk 2]
```diff
@@ -0,0 +1,218 @@
+"
+I am a set of generic tests for the PEG production rules.
+"
+Class {
+	#name : #PP2PEGGrammarTest,
+	#superclass : #TestCase,
```
Comment (kursjan) [inline PetitParser2-PEG/PP2PEGGrammarTest.class.st]: Can't you make this subclass of `PP2AbstractParserTest`?

Comment (kursjan) [reply PetitParser2-PEG/PP2PEGGrammarTest.class.st]: The PP2AbstractParserTest provides methods such as:
#assert:parse:
#assert:fail:

Moreover `PP2CompositeNodeTest` provides support for rules such as:
#parse:rule
#fail:rule

See `PP2ExpressionGrammarTest` and `PP2ExpressionParserTest` for examples.

Or did I miss something that prevents you from doing so?

Comment (merkste) [reply PetitParser2-PEG/PP2PEGGrammarTest.class.st]: Hi, I adapted the tests accordingly. I just overlooked that possibility. However, I do not use #parse:rule: , #fail:rule: , and #parserInstanceFor: as the test structure is to compile parsers for different grammars and check their behavior.

**Review by kursjan — APPROVED** [review]
Thanks!

**Comment by merkste** [comment]
Do you want me to squash this branch into two commits (parser and tests) before merging?

**Pull request merged**